src/XrdSecgsi/XrdSecProtocolgsi.cc

```diff
@@ -5082,12 +5111,22 @@ int XrdSecProtocolgsi::QueryProxy(bool checkcache, XrdSutCache *cache,
                   continue;
                }
             }
-            // Parse the proxy file
-            int nci = (*ParseFile)(pi->out, po->chain);
-            if (nci < 2) {
-               DEBUG("proxy files must have at least two certificates"
-                     " (found: "<<nci<<")");
-               continue;
+            if (pi->nopxy) {
+               // Parse the cert file
+               int nci = (*ParseFile)(pi->cert, po->chain, pi->key);
+               if (nci < 1) {
+                  DEBUG("proxy files must have at least 1 certificates"
```

Comment:
Is this log message correct, aren't we parsing here the certificate (not the proxy)?

Reply:
Correct. Fixed. Thanks!